packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.test.tsx

```diff
@@ -7,45 +7,79 @@ import mockFileEntityData, {
   MOCK_FILE_ENTITY_ID,
   MOCK_FILE_NAME,
 } from '@/mocks/entity/mockFileEntity'
-import { getEntityHandlers } from '@/mocks/msw/handlers/entityHandlers'
-import { server } from '@/mocks/msw/server'
+import { useGetEntityHeaders } from '@/synapse-queries/entity/useGetEntityHeaders'
+import { getUseQueryMock } from '@/testutils/ReactQueryMockUtils'
 import { createWrapper } from '@/testutils/TestingLibraryUtils'
-import { MOCK_REPO_ORIGIN } from '@/utils/functions/getEndpoint'
-import { ReferenceList } from '@sage-bionetworks/synapse-types'
+import { SynapseClientError } from '@sage-bionetworks/synapse-client'
+import {
+  EntityHeader,
+  PaginatedResults,
+  ReferenceList,
+} from '@sage-bionetworks/synapse-types'
 import { act, render, screen, waitFor } from '@testing-library/react'
 import userEvent from '@testing-library/user-event'
 import * as EntityFinderModule from '../EntityFinder/EntityFinderModal'
 import { EntityFinderModal } from '../EntityFinder/EntityFinderModal'
 import { EntityHeaderTable, EntityHeaderTableProps } from './EntityHeaderTable'
 
+vi.mock(
+  '@/synapse-queries/entity/useGetEntityHeaders',
+  async importOriginal => ({
+    ...(await importOriginal()),
+    useGetEntityHeaders: vi.fn(),
+  }),
+)
```

Comment:
Fixed this flaky test by mocking the hook, adding fake timers

With Jest, this always passed in CI, but sometimes failed locally. This failed 100% of the time in Vitest (which is good!) before these changes.